Added image and version variable & resource limits

martijnvdp opened the pull request:
minor changes: 
- Changed token var names to pass tfsec security check
- Added image and version variables so an image on ecr can be used
- enable/added vars to set resource limits
- added option to pass extra resources to the rbac profile of the service account

redeux commented:
Thanks for the contribution! I looked over the changes briefly, and they make sense to me. Let me know when this is ready for a formal review.

martijnvdp commented:
yes this is ready, maybe the default resource values could still be improved , havent put much time in that.

redeux commented on kubernetes_deployment.tf:
```diff
@@ -11,7 +11,7 @@ resource "kubernetes_deployment" "tfc_agent" {
 
   spec {
-    replicas = 1
+    replicas = var.agent_replicacount
```
should this be `agent_replicas` to match the manifest attribute name?

martijnvdp replied:
yes that might be better i've changed it

redeux commented on variables.tf:
```diff
@@ -20,15 +16,26 @@ variable "agent_log_level" {
 }
 
-variable "namespace" {
+variable "agent_replicacount" {
```
Same question about `replicas`

redeux commented on variables.tf:
```diff
@@ -41,4 +48,40 @@ variable "create_namespace" {
+
+variable "limits_memory" {
+  type        = string
+  default     = "2Gi"
```
I'm worried that this is too low.  I'm going to see if I can find sensible requests and limits.

redeux commented:
Thanks for the contribution!  I left just a couple comments.  I'm also going to see if I can find the appropriate specs for the requests and limits.

redeux commented:
It looks like the default limits you put are pretty sensible.

martijnvdp commented:
Thanks for the review , i've renamed the variable and also added cilium resources to the rbac to be able to deploy cilium network policys with terraform using the agent.

redeux commented:
This mostly looks good to me.  A couple more changes I would request.

- Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
- I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
- I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.

redeux reviewed (changes requested):
This mostly looks good to me.  A few more changes I would request.

- Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
- I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
- I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.

martijnvdp commented:
> This mostly looks good to me. A few more changes I would request.
> 
> * Changing the variable names makes this a breaking change, so the version bump will be to `0.1.0`
ok I've changed the version to 0.1.0
> * I'm not in favor of including custom objects in the RBAC policy. I think we can use the [concat function](https://www.terraform.io/docs/language/functions/concat.html) to merge the standard list with any user defined additions.
Yes that makes sense , i've changed this and added optional variables to add additional resources/api-groups
> * I noticed the `agent_token` variable isn't marked sensitive, but it probably should be.
Yes you're right should be sensitive i've changed it
Thanks for the review

redeux reviewed (commented):
The module apply fails if you leave `cluster_access_rbac_api_gourps` or `cluster_acess_rbac_resources` empty.  Looks like everything works fine if the default is changed to `  default     = []`
